engine/src/main/java/org/terasology/rendering/shader/ShaderParametersBase.java

```diff
@@ -43,27 +45,29 @@ public void applyParameters(Material program) {
         program.setFloat("viewingDistance", CoreRegistry.get(Config.class).getRendering().getViewDistance().getChunkDistance().x * 8.0f);
 
         WorldRenderer worldRenderer = CoreRegistry.get(WorldRenderer.class);
-        WorldProvider worldProvider = CoreRegistry.get(WorldProvider.class);
+        BackdropProvider backdropProvider = CoreRegistry.get(BackdropProvider.class);
 
-        if (worldRenderer != null) {
-            program.setFloat("daylight", worldRenderer.getDaylight(), true);
+        if (worldRenderer != null && backdropProvider != null) {
```

Comment:
I don't quite understand these and other checks elsewhere for the WorldRenderer being null. I'm conforming to the norm, but isn't the WorldRenderer something quite major we can assume to be NOT null?